crates/bevy_ecs/src/system/commands/mod.rs

```diff
@@ -930,6 +931,50 @@ impl EntityCommands<'_> {
         self.add(insert(bundle))
     }
 
+    /// Adds a dynamic component to an entity.
+    ///
+    /// See [`EntityWorldMut::insert_by_id`] for more information.
+    ///
+    /// # Panics
+    ///
+    /// The command will panic when applied if the associated entity does not exist.
+    ///
+    /// To avoid a panic in this case, use the command [`Self::try_insert`] instead.
+    ///
+    /// # Safety
+    ///
+    /// - [`ComponentId`] must be from the same world as `self`.
+    /// - `T` must have the same layout as the one passed during `componennt_id` creation.
+    pub unsafe fn insert_by_id<T: Send + 'static>(
+        &mut self,
+        component_id: ComponentId,
+        value: T,
+    ) -> &mut Self {
+        // SAFETY: same invariants as parent call
+        self.add(unsafe {insert_by_id(component_id, value, move |entity| {
+            panic!("error[B0003]: Could not insert a component {component_id:?} (with type {}) for entity {:?} because it doesn't exist in this World. See: https://bevyengine.org/learn/errors/#b0003", std::any::type_name::<T>(), entity);
```

Comment:
nitpick: Use inline-formatting for `entity` as well.

```suggestion
            panic!("error[B0003]: Could not insert a component {component_id:?} (with type {}) for entity {entity:?} because it doesn't exist in this World. See: https://bevyengine.org/learn/errors/#b0003", std::any::type_name::<T>());
```

suggestion(non-blocking, preference): Use a named parameter for the type_name, since the format string is kinda long and also has inline-parameters as well:
```suggestion
            panic!("error[B0003]: Could not insert a component {component_id:?} (with type {component_type_name}) for entity {:?} because it doesn't exist in this World. See: https://bevyengine.org/learn/errors/#b0003", component_type_name = std::any::type_name::<T>(), entity);
```